public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -131,6 +131,23 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     return of({ state: LoadingState.Done, data: [] });
   }
 
+  targetContainsTemplate(query: AzureMonitorQuery) {
+    if (query.subscription && this.templateSrv.variableExists(query.subscription)) {
+      return true;
+    }
+
+    let subQuery;
+    if (query.queryType === AzureQueryType.AzureMonitor) {
+      subQuery = JSON.stringify(query.azureMonitor);
+    } else if (query.queryType === AzureQueryType.LogAnalytics) {
+      subQuery = JSON.stringify(query.azureLogAnalytics);
+    } else if (query.queryType === AzureQueryType.AzureResourceGraph) {
+      subQuery = JSON.stringify([query.azureResourceGraph, query.subscriptions]);
```

Comment:
can you add a test that covers the case of a variable in this `subscriptions`? and maybe move the condition to check `query.subscriptions` to it's own `if`

Reply:
Yeah good catch!